src/background_script.js

```diff
@@ -70,6 +70,31 @@ function setupOnBeforeSendHeaders() {
   );
 }
 
+function updateBadgeStatus(currentTab) {
+  const currentHostname = new URL(currentTab.url).hostname;
+  const urls = [...enabledHostnames.get_values()];
+
+  if (urls.includes(currentHostname)) {
+    browser.browserAction.setIcon({
+      path: "assets/badge-indicator-on.svg" });
+    browser.browserAction.setTitle({ title: "Chrome Mask is on." });
+  } else {
+    browser.browserAction.setIcon({
+      path: "assets/badge-indicator-off.svg" });
+    browser.browserAction.setTitle({ title: "Chrome Mask is off." });
+  }
```

Comment:
In theory, you could replace the repetition here with a

```js
const state = urls.includes(currentHostname) ? "on" : "off";
browser.browserAction.setIcon({
  path: `assets/badge-indicator-${state}.svg`,
});
browser.browserAction.setTitle({ title: `Chrome Mask is ${state}.` });
```

but I'm not gonna block on that. If you decide to leave the current implementation, please at least run `npm run lint` on it, so that prettier can fix your formatting. :)

Reply:
At first, I had the same idea. However, I'd like to i18n the addon and I didn't remember if the API supports that use case (passing a variable).

Reply:
Makes sense, let's dismiss that, then.